src/common/utils.js

```diff
@@ -590,3 +592,61 @@ export const createFakeProgress = (() => {
 export const ShortDate = ({ timestamp }) => (
   <FormattedDate value={timestamp} month='short' day='numeric' year='numeric' />
 )
+
+// ===================================================================
+
+// Smart backup
+export const destructPattern = (pattern, valueTransform = identity) =>
+  pattern && {
+    not: !!pattern.__not,
+    values: valueTransform((pattern.__not || pattern).__or),
+  }
+
+export const constructPattern = (
+  { not, values } = EMPTY_OBJECT,
+  valueTransform = identity
+) => {
+  if (values == null || !values.length) {
+    return
+  }
+
+  const pattern = { __or: valueTransform(values) }
+  return not ? { __not: pattern } : pattern
+}
+
+const parsePattern = pattern => {
+  const patternValues = map(pattern.values, value => {
+    return isArray(value) ? new CM.String(value[0]) : new CM.String(value)
+  })
+
+  return pattern.not
+    ? new CM.Not(new CM.Or(patternValues))
+    : new CM.Or(patternValues)
+}
+
+export const constructFilter = pattern => {
```

Comment:
Now that these functions are exposed in `utils`, I think we have to find better names for them.